### ISSUE TITLE
LDAP group sync is not working

### ISSUE THREAD

**Issue by tograss** [goauthentik/authentik#6506]
**Describe the bug**
When using a LDAP source to sync users group changes on the LDAP source are not reflected in authentic. 

**To Reproduce**
1. Create LDAP Source
2. Sync Users
3. Change group(s) of a user
4. Sync again
-> user keeps the old group memberships

**Expected behavior**
The groups should be updated

**Version and Deployment (please complete the following information):**

-   authentik version: 2023.6.1
-   Deployment: docker-compose

**Additional context**
I think this worked before the upgrade to 2023.6

**Comment by tograss** [comment]
I think I found the culprit. During user sync the following error occurs:
```

1691580562.7152789 [error    ] Task authentik.sources.ldap.tasks.ldap_sync_single[cd4eb723-c23d-49ea-9392-675823352f48] raised unexpected: RuntimeError('Never call result.get() within a task!\nSee https://docs.celeryq.dev/en/latest/userguide/tasks.html#avoid-launching-synchronous-subtasks\n') [celery.app.trace]
  Traceback (most recent call last)
 lib/python3.11/site-packages/celery/app/trace.py:477 in trace_task

   474                if task_before_start:
   475                   task_before_start(uuid, args, kwargs)
   476
   477                R = retval = fun(*args, **kwargs)
   478                state = SUCCESS
   479             except Reject as exc:
   480                I, R = Info(REJECTED, exc), ExceptionInfo(internal=True)

  authentik/sources/ldap/tasks.py:53 in ldap_sync_single

    50          ldap_sync_paginator(source, MembershipLDAPSynchronizer),
    51       ),
    52    )
    53    task()
    54
    55
    56 def ldap_sync_paginator(source: LDAPSource, sync: type[BaseLDAPSynchronizer]) -> list:

 /lib/python3.11/site-packages/celery/canvas.py:2038 in __call__

   2035       self.subtask_type = 'chord'
   2036
   2037    def __call__(self, body=None, **options):
   2038       return self.apply_async((), {'body': body} if body else {}, **options)
   2039
   2040    def __or__(self, other):
   2041       if (not isinstance(other, (group, _chain)) and

 /lib/python3.11/site-packages/celery/canvas.py:2131 in apply_async

   2128              else group(self.tasks, app=app, task_id=self.options.get('task_id', uui
   2129       if app.conf.task_always_eager:
   2130          with allow_join_result():
   2131             return self.apply(args, kwargs,
   2132                           body=body, task_id=task_id, **options)
   2133
   2134       merged_options = dict(self.options, **options) if options else self.options

 /lib/python3.11/site-packages/celery/canvas.py:2150 in apply

   2147       tasks = (self.tasks.clone() if isinstance(self.tasks, group)
   2148              else group(self.tasks, app=self.app))
   2149       return body.apply(
   2150          args=(tasks.apply(args, kwargs).get(propagate=propagate),),
   2151       )
   2152
   2153    @classmethod

 /lib/python3.11/site-packages/celery/result.py:701 in get

    698       in addition it uses :meth:`join_native` if available for the
    699       current result backend.
    700       """
    701       return (self.join_native if self.supports_native_join else self.join)(
    702          timeout=timeout, propagate=propagate,
    703          interval=interval, callback=callback, no_ack=no_ack,
    704          on_message=on_message, disable_sync_subtasks=disable_sync_subtasks,

 /lib/python3.11/site-packages/celery/result.py:756 in join

    753             seconds.
    754       """
    755       if disable_sync_subtasks:
    756          assert_will_not_block()
    757       time_start = time.monotonic()
    758       remaining = None
    759

 /lib/python3.11/site-packages/celery/result.py:37 in assert_will_not_block

     34
     35 def assert_will_not_block():
     36    if task_join_will_block():
     37       raise RuntimeError(E_WOULDBLOCK)
     38
     39
     40 @contextmanager
RuntimeError: Never call result.get() within a task!
See https://docs.celeryq.dev/en/latest/userguide/tasks.html#avoid-launching-synchronous-subtasks

```

This stops the sync, which leads to groups and memberships not being synced.

**Comment by medir** [comment]
We are using the Docker version and we are experiencing the same problem. We upgraded from version 2023.5.5 (the synchronization of users, groups and memberships worked fine) to the version 2023.6.1 and it stopped working. We tried to downgrade to version 2023.6.0 and it does not synchronize neither groups and memberships. So it look like there is a bug introduced in version 2023.6.0.

**Comment by MaximeWewer** [comment]
Hello,
I have the same problem on versions 2023.6. On versions 2023.5 and before no problem. @BeryJu, are the changes to LDAP on 2023.6 releases causing these issues ?

**Comment by tograss** [comment]
This is fixed in 8.1. At least for me
